test/Microsoft.ML.OnnxTransformerTest/OnnxTransformTests.cs

```diff
@@ -332,6 +332,78 @@ public void OnnxModelMultiInput()
             }
         }
 
+        [OnnxFact]
+        public void OnnxModelOutputDifferentOrder()
+        {
+            var modelFile = Path.Combine(Directory.GetCurrentDirectory(), "twoinput", "twoinput.onnx");
+            var env = new ConsoleEnvironment(seed: 1);
+            var samplevector = GetSampleArrayData();
+
+            var dataView = ML.Data.LoadFromEnumerable(
+                new TestDataMulti[] {
+                    new TestDataMulti()
+                    {
+                        ina = new float[] {1,2,3,4,5},
+                        inb = new float[] {1,2,3,4,5}
+                    }
+                });
+            // The model returns the output columns in the order outa, outb. We are doing the opposite here, making sure the name mapping is correct.
+            var onnx = ML.Transforms.ApplyOnnxModel(new[] { "outb", "outa" }, new[] { "ina", "inb" }, modelFile).Fit(dataView).Transform(dataView);
+
+            var outaCol = onnx.Schema["outa"];
+            var outbCol = onnx.Schema["outb"];
+            using (var curs = onnx.GetRowCursor(outaCol, onnx.Schema["outb"]))
+            {
+                var getScoresa = curs.GetGetter<VBuffer<float>>(outaCol);
+                var getScoresb = curs.GetGetter<VBuffer<float>>(outbCol);
```

Comment:
_(appended to code to allow threaded discussion)_

So I don't remember correctly, but today during scrum my impression was that the code ran correctly if you used a Preview() of the output DataView, and it only failed if you tried to use a PredictionEngine with an output class. Is this true?

Reply:
No, it fails if the order of the output columns you provide is different then the order they actually are in the ONNX model.

Reply:
And it does it regardless of how the columns are retrieved. It complains about the data types with the getter.